src/libsyntax/ast.rs

```diff
@@ -456,7 +472,7 @@ pub enum MetaItemKind {
     /// List meta item.
     ///
     /// E.g. `derive(..)` as in `#[derive(..)]`
-    List(InternedString, Vec<P<MetaItem>>),
+    List(InternedString, Vec<MetaListItem>),
```

Comment:
It's preferable to benchmark before landing, because otherwise it will mean immediately landing a second PR just to fix it. `rustup` will let you use a local build as the default; you can build `serde` and run its perf benchmarks with both.